src/core/qgsreadwritecontext.h

```diff
@@ -31,26 +31,75 @@ class CORE_EXPORT QgsReadWriteContext
 {
   public:
 
+    /**
+     * Struct for QgsReadWriteContext error or warning messages
+     * \since QGIS 3.2
+     */
+    struct ReadWriteMessage
+    {
+        //! Construct a container for QgsReadWriteContext error or warning messages
+        ReadWriteMessage( const QString &message, Qgis::MessageLevel level = Qgis::Warning,  QStringList categories = QStringList() )
+          : mMessage( message )
+          , mLevel( level )
+          , mCategories( categories )
+        {}
+
+        //! Return the message string
+        QString message() const {return mMessage;}
+
+        //! Return the message level
+        Qgis::MessageLevel level() const {return mLevel;}
+
+        //! Return the stack of categories of the message
+        QStringList categories() const {return mCategories;}
+
+      private:
+        QString mMessage = QString();
```

Comment:
no need for the =QString(), =QStringList() here

Reply:
fixed